pkg/cmd/create/create_devpod.go

```diff
@@ -107,6 +108,72 @@ type CreateDevPodOptions struct {
 	Results CreateDevPodResults
 }
 
+// devPodLabels split gitInfo labels
+type devPodLabels struct {
+	gitSchemeLabelKey   string
+	gitSchemeLabelValue string
+	gitHostLabelKey     string
+	gitHostLabelValue   string
+	gitOrgLabelKey      string
+	gitOrgLabelValue    string
+	gitRepoLabelKey     string
+	gitRepoLabelValue   string
+
+	gitLabels map[string]string
+}
+
+// populateFromGitInfo populate this struct with the git repository elements (scheme, host, org and repo)
+func (d *devPodLabels) populateFromGitInfo(gitInfo *gits.GitRepository) error {
+	d.gitSchemeLabelKey = naming.ToValidNameTruncated(fmt.Sprintf("%s/scheme", kube.LabelDevPodGitPrefix), 63)
+	d.gitSchemeLabelValue = naming.ToValidNameTruncated(gitInfo.Scheme, 63)
+	d.gitHostLabelKey = naming.ToValidNameTruncated(fmt.Sprintf("%s/host", kube.LabelDevPodGitPrefix), 63)
+	d.gitHostLabelValue = naming.ToValidNameTruncated(gitInfo.Host, 63)
+
+	d.gitOrgLabelKey = fmt.Sprintf("%s/organisation", kube.LabelDevPodGitPrefix)
```

Comment:
I think this will create an invalid label key with to '/':

```
error: finding DevPod by selector: map[jenkins-io-repo-host:github-com jenkins-io-repo-scheme:https jenkins.io/devpod_user:hardy jenkins.io/repo/name:go-eks jenkins.io/repo/organisation:hf-bee]: converting label selector: map[jenkins-io-repo-host:github-com jenkins-io-repo-scheme:https jenkins.io/devpod_user:hardy jenkins.io/repo/name:go-eks jenkins.io/repo/organisation:hf-bee]: invalid label key "jenkins.io/repo/organisation": a qualified name must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]') with an optional DNS subdomain prefix and '/' (e.g. 'example.com/MyName')
```

I think we need to create new labels.

Reply:
You are right. We must sanitize the label keys (I use now the ToValidNameTruncated function to all labels keys now)